components/scream/src/physics/shoc/shoc_functions_f90.cpp

```diff
@@ -495,5 +507,69 @@ void shoc_diag_second_moments_srf_f(Int shcol, Real* wthl, Real* uw, Real* vw, R
   ekat::pack::device_to_host({ustar2, wstar}, shcol, out_views);
 }
 
+void shoc_diag_second_moments_lbycond_f(Int shcol, Real* wthl, Real* wqw, Real* uw, Real* vw, Real* ustar2, Real* wstar,
+                        Real* wthlo, Real* wqwo, Real* uwo, Real* vwo, Real* wtkeo, Real* thlo, Real* qwo, Real* qwthlo)
+{
+  using SHOC       = Functions<Real, DefaultDevice>;
+  using Scalar     = typename SHOC::Scalar;
+  using Pack1      = typename ekat::pack::Pack<Real, 1>;
+  using view_1d    = typename SHOC::view_1d<Pack1>;
+
+  Kokkos::Array<view_1d, 6> lbycond_d;
+  ekat::pack::host_to_device({wthl, wqw, uw, vw, ustar2, wstar}, shcol, lbycond_d);
+
+  // inputs
+  view_1d wthl_d  (lbycond_d[0]),
+          wqw_d   (lbycond_d[1]),
+          uw_d    (lbycond_d[2]),
+          vw_d    (lbycond_d[3]),
+          ustar2_d(lbycond_d[4]),
+          wstar_d (lbycond_d[5]);
+
+  // outputs
+  view_1d wthlo_d  ("wthl", shcol),
+          wqwo_d   ("wqw" , shcol),
+          uwo_d    ("uw"  , shcol),
+          vwo_d    ("vw"  , shcol),
+          wtkeo_d  ("wtke", shcol),
+          thlo_d   ("thl" , shcol),
+          qwo_d    ("qw"  , shcol),
+          qwthlo_d ("qwthl", shcol);
+
+  Kokkos::parallel_for("parallel_moments_lbycond", shcol, KOKKOS_LAMBDA (const int& i) {
```

Comment:
Curious what folks think about using `k` as the vertical index by convention, since we often talk about `i` as being the column index. Just a thought.

Reply:
@jeff-cohere , he's iterating over schol (across columns), so I think `i` makes sense here.